core/cpu.cpp

```diff
@@ -618,9 +595,352 @@ u8 CPU::exec(){
             op_cp(a, *r);
             return 4;
         }
+        case 0xC0: {
+            //RET NZ
+            if(readSubtractFlag() & readZeroFlag()){
+                pc = popFromStack();
+                return 20;
+            }
+            else{
+                return 8;
+            }
+        }
+        case 0xC2: {
+            //JP NZ, a16
+            //check if subtraction flag(n) & zero flags are set
+
+            //PC NEEDS TO BE INCREMENTED TWICE ON 16 BIT READ
+            u16 nn_nn = mmu.read16Bit(pc++);
+            pc++;
+
+            if(readZeroFlag() & readSubtractFlag()){
+                pc = nn_nn;
+                return 16;
+            }
+            else{
+                return 12;
+            }
+        }
+        case 0xC3: {
+            //JP a16
+            //PC NEEDS TO BE INCREMENTED TWICE ON 16 BIT READ
+            u16 nn_nn = mmu.read16Bit(pc++);
+            pc++;
+            pc = nn_nn;
+            return 16;
+        }
+        case 0xC4: {
+            //CALL NZ, a16
+            //check if subtraction flag(n) & zero flags are set
+
+            //PC NEEDS TO BE INCREMENTED TWICE ON 16 BIT READ
+            u16 nn_nn = mmu.read16Bit(pc++);
+            pc++;
+
+            if(readZeroFlag() & readSubtractFlag()){
+                pushToStack(pc);
+                pc = nn_nn;
+                return 24;
+            }
+            else{
+                return 12;
+            }
+        }
+        case 0xC6: {
+            //ADD A, d8
+            u8 n = mmu.read(pc++);
+            u8 a = getHighByte(af);
+            u8 result = op_add(a, n);
+            setHighByte(&af, result);
+
+            return 4;
+        }
+        case 0xC7: case 0xCF: case 0xD7: case 0xDF: case 0xE7: case 0xEF: case 0xF7: case 0xFF: {
+            //RST 00H, 08H, 10H, 18H, 20H, 28H, 30H, 38H
+            pushToStack(pc);
+
+            u8 call_value = (opCode-0xC7);
```

Comment:
I thought this was a clever way to generalize the implementation of these op codes.